### PR TITLE
Refactor fulfillment details page

### PR DESCRIPTION
**Pull request by Droniu**
I want to merge this change because it refactors fulfillment details page for usage with single warehouse context.

**PR intended to be tested with API branch:**  3.1

### Screenshots
Before:
image

After:
3rd item on the list is preorder case
image

### Pull Request Checklist

1. [x] This code contains UI changes
2. [x] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [x] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://master.staging.saleor.cloud/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-4790 Refactor Fulfillment details page](https://app.clickup.com/t/2549495/SALEOR-4790)

**Comment by taniotanio7** [comment]
Can't open fulfill page - due to an error in the query `WarehouseDetails`

CleanShot 2022-04-01 at 14 31 35@2x
CleanShot 2022-04-01 at 14 32 08@2x

**Comment by Droniu** [comment]
@taniotanio7 this is bug introduced in #1875. For some reason default warehouse selection fails, I'm gonna fix this in future PR, or maybe in this, we'll see.
Workaround for now -> manually choose warehouse before clicking "fulfill". It's this chevron in top right of unfulfilled card -> 
image

**Review by dominik-zeglen — APPROVED** [review]

**Comment by taniotanio7** [comment]
When the selected warehouse doesn't have the item to fulfill in stock the value in "Stock" column is "NaN"

CleanShot 2022-04-01 at 15 07 24@2x
CleanShot 2022-04-01 at 15 07 33@2x

File: src/orders/components/OrderFulfillPage/utils.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { OrderFulfillDataQuery } from "@saleor/graphql";
+
+export const getDefaultFulfillmentValue = (
+  line: OrderFulfillDataQuery["order"]["lines"][0],
```
Comment (taniotanio7) [inline src/orders/components/OrderFulfillPage/utils.ts:4]: avoid deeply nested types, extract to fragment and then import that fragment's type

example: 
```ts
// src/fragments/errors.ts:266
export const voucherBulkDeleteError = gql`
  fragment VoucherBulkDeleteError on DiscountError {
    code
    field
    message
  }
`;
```
then import from types.generated.ts

```ts
export type VoucherBulkDeleteErrorFragment = { __typename: 'DiscountError', code: DiscountErrorCode, field: string | null, message: string | null };
```

File: src/orders/components/OrderFulfillLine/OrderFulfillLine.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+        >
+          <TextField
+            type="number"
+            inputProps={{
+              className: classNames(classes.quantityInnerInput, {
+                [classes.quantityInnerInputNoRemaining]: !line.variant
+                  ?.trackInventory
+              }),
+              min: 0,
+              style: { textAlign: "right" }
```
Comment (taniotanio7) [inline src/orders/components/OrderFulfillLine/OrderFulfillLine.tsx:121]: Could this use a `className` instead?

**Review by taniotanio7 — COMMENTED** [review]
+ this comment https://github.com/saleor/saleor-dashboard/pull/1915#issuecomment-1085890753

File: src/orders/components/OrderFulfillLine/OrderFulfillLine.tsx (hunk 2 above)
Comment (Droniu) [reply src/orders/components/OrderFulfillLine/OrderFulfillLine.tsx:121]: No, it's MUI magic.

**Review by taniotanio7 — APPROVED** [review]

**Pull request merged**